### PR TITLE
[move] Update the move commit and the generated document

### PR DESCRIPTION
**Pull request by junkil-park**
### Description

This PR updates the move commit to bring the cherry-pick commit (https://github.com/move-language/move/pull/639). It brings the fixes on the Prover, the document generator and the compiler.

This PR also updates the Z3 version which is the dependency of Prover.

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/5380)

### CODE REVIEW

File: aptos-move/framework/src/built_package.rs [hunk 1]
```diff
@@ -93,7 +93,7 @@ pub(crate) fn build_model(
         test_mode: false,
         force_recompilation: false,
         fetch_deps_only: false,
-        fetch_latest_git_deps: false,
+        skip_fetch_latest_git_deps: false,
```
Comment (junkil-park) [inline aptos-move/framework/src/built_package.rs:96]: @gregnazario , @wrwg , I had to change the name from `fetch_latest_git_deps` to `skip_fetch_latest_git_deps` to reflect https://github.com/move-language/move/pull/588. I'd like to make sure the initial value here (`false`) is still correct?

File: aptos-move/framework/src/built_package.rs [hunk 2]
```diff
@@ -120,7 +120,7 @@ impl BuiltPackage {
             test_mode: false,
             force_recompilation: false,
             fetch_deps_only: false,
-            fetch_latest_git_deps: false,
+            skip_fetch_latest_git_deps: false,
```
Comment (junkil-park) [inline aptos-move/framework/src/built_package.rs:123]: @gregnazario , @wrwg , here is another one.

File: aptos-move/framework/src/built_package.rs (hunk 1 above)
Comment (wrwg) [reply aptos-move/framework/src/built_package.rs:96]: Recommend to set this to true without guidance from @gregnazario so we do not break anything here. @davidiw any different take?

**Review by wrwg — APPROVED** [review]

File: aptos-move/framework/src/built_package.rs (hunk 1 above)
Comment (junkil-park) [reply aptos-move/framework/src/built_package.rs]: OK. I set it to be true.

File: aptos-move/framework/src/built_package.rs (hunk 2 above)
Comment (junkil-park) [reply aptos-move/framework/src/built_package.rs]: OK. I set this to be true as well.

**Review by rahxephon89 — APPROVED** [review]

**Pull request merged**